### DISCUSSION TITLE
About print information bug

### DISCUSSION THREAD

**Discussion by adrian-yehe** [gabime/spdlog#2558] [Q&A] [answered]
The printed information is different from the expected result。
code: 
`
#define SMMU_VER_MAJOR   (3)
#define SMMU_VER_MINOR   (0)
#define SMMU_VER_PATCH   (1)
spdlog::info("Welcome to SMMU version 0x{0:x}.{0:d}.0x{0:x}  !", SMMU_VER_MAJOR, SMMU_VER_MINOR, SMMU_VER_PATCH);
`
result:
>[2022-11-29 16:04:12.577] [info] Welcome to SMMU version 0x3.3.0x3  !
but i want to show information below:
>[2022-11-29 16:04:12.577] [info] Welcome to SMMU version 0x3.3.0x01  !

**Comment by tt4g** [comment, marked as answer]
Your syntax of fmt library is wrong. Replace last syntax `0x{0:x}` with `0x{:#02d}` for padding left.

```diff
-#define SMMU_VER_MAJOR (3) #define SMMU_VER_MINOR (0) #define SMMU_VER_PATCH (1) spdlog::info("Welcome to SMMU version 0x{0:x}.{0:d}.0x{0:x} !", SMMU_VER_MAJOR, SMMU_VER_MINOR, SMMU_VER_PATCH);
+#define SMMU_VER_MAJOR (3) #define SMMU_VER_MINOR (0) #define SMMU_VER_PATCH (1) spdlog::info("Welcome to SMMU version 0x{0:x}.{0:d}.0x{:#02d} !", SMMU_VER_MAJOR, SMMU_VER_MINOR, SMMU_VER_PATCH);
```

See fmt syntax: https://fmt.dev/9.0.0/syntax.html

**Reply by adrian-yehe** [reply]
thanks; should be written like this:
>#define SMMU_VER_MAJOR (3) #define SMMU_VER_MINOR (0) #define SMMU_VER_PATCH (1) spdlog::info("Welcome to SMMU version 0x{0:x}.{1:d}.0x{2:d} !", SMMU_VER_MAJOR, SMMU_VER_MINOR, SMMU_VER_PATCH);
